clients/src/test/java/org/apache/kafka/common/record/MemoryRecordsTest.java

```diff
@@ -447,8 +444,8 @@ public void testBuildEndTxnMarker() {
 
     @Test
     public void testFilterToBatchDiscard() {
-        if (compression != CompressionType.NONE || magic >= RecordBatch.MAGIC_VALUE_V2) {
-            expectExceptionWithZStd(compression, magic);
+        if ((compression != CompressionType.NONE && compression != CompressionType.ZSTD) ||
```

Comment:
If we're starting to use `Assume`, this seems like it should be one since the condition just skips the whole test?

Reply:
I was thinking the same, but didn't do it in the end. Will change.